### PR TITLE
Fixing default property, adding latency test

### PR DESCRIPTION
**Pull request by davidartplus**

### CODE REVIEW

File: micro-s3/build.gradle [hunk 1]
```diff
@@ -5,6 +5,10 @@ dependencies {
 	compile 'com.amazonaws:aws-java-sdk:' + s3Version
 	compile project(':micro-core')
 	compile project(':micro-manifest-comparator')
+    compile group: 'org.springframework', name: 'spring-test', version: '4.0.5.RELEASE'
```
Comment (johnmcclean-aol) [inline micro-s3/build.gradle]: these should probably be testCompile by the looks of things.
Also there is a top level gradle.properties file where dependency versions can be specified. Keeps them the same across all included plugins.

Comment (davidartplus) [inline micro-s3/build.gradle]: Fixed

File: micro-s3/src/main/java/com/aol/micro/server/s3/data/ReadUtils.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.s3.transfer.Download;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.aol.cyclops.util.ExceptionSoftener;
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class ReadUtils {
+    private TransferManager transferManager;
```
Comment (johnmcclean-aol) [inline micro-s3/src/main/java/com/aol/micro/server/s3/data/ReadUtils.java]: Can we make these final or are the mutable?

Comment (davidartplus) [inline micro-s3/src/main/java/com/aol/micro/server/s3/data/ReadUtils.java]: yep, final.

File: micro-s3/src/main/java/com/aol/micro/server/s3/data/S3Utils.java [hunk 3]
```diff
@@ -43,23 +33,25 @@
     private final String tmpDirectory;
     private final ExecutorService uploaderService;
     private final boolean aes256Encryption;
+    private final ReadUtils readUtils;
 
     @Autowired
     public S3Utils(AmazonS3Client client, TransferManager transferManager,
             @Value("${s3.tmp.dir:#{systemProperties['java.io.tmpdir']}}") String tmpDirectory,
-            @Value("${s3.aes256.enabled:false}}") boolean aes256Encryption,
+            @Value("${s3.aes256.enabled:false}") boolean aes256Encryption,
```
Comment (morrowgi) [inline micro-s3/src/main/java/com/aol/micro/server/s3/data/S3Utils.java]: Is there any other forms of encryption?

File: micro-s3/src/test/java/com/aol/micro/server/s3/data/S3DownloadSystemTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        HttpTransport httpTransport = new HttpTransport.Builder().withApiKey(System.getProperty("datadog.key"))
+                                                                 .build();
+        DatadogReporter.Builder builder = DatadogReporter.forRegistry(metricsRegistry)
+                                                         .withTransport(httpTransport)
+                                                         .withExpansions(expansions)
+                                                         .withTags(Arrays.asList("stage:dev"));
+        DatadogReporter reporter = builder.build();
+        int reportPeriod = 5;
+        TimeUnit reportUnit = TimeUnit.valueOf("SECONDS");
+        log.info("Reporting to datadog every {} {}", 5, reportUnit);
```
Comment (AolCodingCamp) [inline micro-s3/src/test/java/com/aol/micro/server/s3/data/S3DownloadSystemTest.java]: Should 5 not be reportPeriod ?

Comment (davidartplus) [inline micro-s3/src/test/java/com/aol/micro/server/s3/data/S3DownloadSystemTest.java]: yep

File: micro-s3/src/main/java/com/aol/micro/server/s3/data/S3Utils.java (hunk 3 above)
Comment (davidartplus) [inline micro-s3/src/main/java/com/aol/micro/server/s3/data/S3Utils.java]: not as of now

**Pull request merged**